### PR TITLE
Add `Effect::Dim`

### PR DESCRIPTION
**Pull request by arxanas**
The effect is more useful on a different background color (e.g. on black background and white foreground, the dim effect produces gray text).

Tested using this code:

```rust
use cursive::theme::Effect;
use cursive::utils::markup::StyledString;
use cursive::views::{LinearLayout, TextView};
use cursive::CursiveRunnable;

fn main() {
    let backends: &[(&str, fn() -> CursiveRunnable)] = &[
        ("ncurses", cursive::ncurses),
        ("pancurses", cursive::pancurses),
        ("crossterm", cursive::crossterm),
        ("termion", cursive::termion),
    ];
    for (backend_name, backend) in backends {
        let mut siv = backend().into_runner();
        siv.add_layer(
            LinearLayout::vertical()
                .child(TextView::new(format!(
                    "This is normal text from backend: {}",
                    backend_name
                )))
                .child(TextView::new({
                    let mut s = StyledString::styled("This is bold text ", Effect::Bold);
                    s.append_plain("followed by normal text");
                    s
                }))
                .child(TextView::new({
                    let mut s = StyledString::styled("This is dim text ", Effect::Dim);
                    s.append_plain("followed by normal text");
                    s
                })),
        );
        siv.run();
    }
}
```

It produced the following output (no effect for ncurses/pancurses):

ncurses test (not rendered)
pancurses test (not rendered)
crossterm test (successfully rendered)
termion test (successfully rendered)

### CODE REVIEW

**Comment by arxanas** [comment]
The build failure looks transient, but I don't know how to re-run the build.

File: cursive/src/backends/curses/n.rs [hunk 1]
```diff
@@ -380,6 +381,7 @@ impl backend::Backend for Backend {
         let style = match effect {
             Effect::Reverse => ncurses::A_REVERSE(),
             Effect::Simple => ncurses::A_NORMAL(),
+            Effect::Dim => ncurses::A_DIM(), // @nocommit test
```
Comment (gyscos) [inline cursive/src/backends/curses/n.rs:384]: `@nocommit test`?

**Comment by gyscos** [comment]
Hi, and thanks for the work!

This looks great - just a minor remark regarding a leftover annotation, and it should be good to go! Thanks again!

File: cursive/src/backends/curses/n.rs (hunk 1 above)
Comment (arxanas) [reply cursive/src/backends/curses/n.rs]: Removed.

**Pull request merged**